sdk/messaging/azservicebus/client.go

```diff
@@ -29,14 +36,25 @@ type Client struct {
 // For an example option, see `WithConnectionString`
 type ClientOption func(client *Client) error
 
-// WithConnectionString configures a namespace with the information provided in a Service Bus connection string
+// WithConnectionString configures a namespace to authenticate using a Service Bus connection string
 func WithConnectionString(connStr string) ClientOption {
 	return func(client *Client) error {
 		client.config.connectionString = connStr
 		return nil
 	}
 }
 
+// WithTokenCredential configures a namespace to connect using a TokenCredential.
+// fullyQualifiedNamespace is the Service Bus namespace name (ex: myservicebus.servicebus.windows.net)
+// tokenCredential is one of the credentials in the `github.com/Azure/azure-sdk-for-go/sdk/azidentity` package.
+func WithTokenCredential(fullyQualifiedNamespace string, tokenCredential azcore.TokenCredential) ClientOption {
```

Comment:
Seems the parameter name could be shorter without creating confusion:
```suggestion
func WithTokenCredential(fullyQualifiedNamespace string, credential azcore.TokenCredential) ClientOption {
```

Reply:
Done